### DISCUSSION TITLE
Como vamos prosseguir no desenvolvimento do app mobile

### DISCUSSION THREAD

**Discussion by Rodrigo001-dev** [diego3g/rsxp-2023#95] [Q&A] [answered]
Fala galera!

[Nessa issue sobre a configuração do Detox](https://github.com/diego3g/rsxp-2023/issues/59), surgiu a possiblidade de utilizarmos o Expo Dev Clients (Devolopment Builds) ao invés do Expo Go (Managed Workflow). 

Mas se escolhermos utilizar o Expo Dev Clients todos serão obrigados a gerar builds com as novas libs nativas que forem adicionadas e o Expo Go não irá funcionar mais quando utilizarmos alguma lib incompatível.

Então por isso queria a opinião de vocês sobre como deveremos seguir 😃

**Comment by orodrigogo** [comment, marked as answer]
Olá @Rodrigo001-dev,

Exatamente. Surgiu essa possibilidade, uma vez que, por enquanto o Detox não roda no Expo Go. A princípio, os testes E2E são executados em cima da build (os chamados black box testing).

Mas, não tem problema se precisarmos seguir com Devolopment Builds. A experiência de desenvolvimento vai ser próxima da que já estamos trabalhamos. 

O que mudaria é: 
- Precisamos ter o ambiente nativo configurado (Android e/ou iOS). Nessa doc tem todo o passo à passo para configurar o seu ambiente: https://react-native.rocketseat.dev/. Uma vez configurado, não precisa mais se preocupar com o ambiente.
- Sempre que houver modificação em código nativo, é necessário gerar uma nova dev build. As builds locais são bem rápida e tranquilas;
- Utilizar o Expo Dev Clients, que nada mais é, um Expo Go customizado que utiliza o código nativo dessa build que geramos e disponibilizamos no dispositivo.

Isso permite continuar utilizando as API padrão do próprio Expo, e utilizar as que ele não tem suporte padrão.

Espero ter esclarecido, e qualquer coisa, conta comigo! 💜

**Reply by Rodrigo001-dev** [reply]
Show! Ficou bem claro para mim, muito obrigado

**Comment by Rodrigo001-dev** [comment]
Vou fechar a discussão, pois consegui entender o que devo fazer!
